sgt-back-end/index.js

```diff
@@ -0,0 +1,164 @@
+const express = require('express');
+const pg = require('pg');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    SELECT
+      *
+    FROM
+      grades;
+  `;
+
+  db.query(sql)
+    .then(data => {
+      const grades = data.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      // eslint-disable-next-line no-console
+      console.log('Get error', err.message);
+      res.status(500).json({ error: 'An unexpected error occurred' });
+    });
+
+});
+
+app.post('/api/grades', (req, res) => {
+  const { course, name, score } = req.body;
+  const numScore = Number(score);
+  if (!Number.isInteger(numScore) || numScore > 100 || numScore < 0) {
+    res.status(400).json({ error: 'Score must be an integer' });
+    return;
+  }
+  if (score === '') {
+    res.status(400).json({ error: 'No score' });
+    return;
+  }
+  if (course === '') {
+    res.status(400).json({ error: 'No course' });
+    return;
+  }
+  if (name === '') {
+    res.status(400).json({ error: 'No name' });
+    return;
+  }
```

Comment:
These will not catch `undefined` if one of them is excluded in the request. A better solution would be to use `if(!course)`,  same style for both.